rbac/api/cross_access/util.py

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Handler for cross-account request clean up."""
+import logging
+
+from django.db.models import Q
+from django.utils import timezone
+from prometheus_client import Summary
+from tenant_schemas.utils import tenant_context
+
+from api.models import CrossAccountRequest, Tenant
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+# Create processing time metric
+PROCESSING_TIME = Summary(
+    "cross_account_expiry_processing_second", "Time spent checking and expiring cross-account requests"
+)
+
+
+@PROCESSING_TIME.time()
+def check_cross_request_expiry():
+    """Check if a cross-account requests have expired, tag them if so."""
+    with tenant_context(Tenant.objects.get(schema_name="public")):
+        expired_cars = []
+        cars = CrossAccountRequest.objects.filter(Q(status="pending") | Q(status="approved"))
+        logger.info("Running expiry check on %d cross-account requests.", len(cars))
+        for car in cars:
+            logger.debug("Checking for expiration of cross-account request %s.", car.pk)
+            if car.end_date < timezone.now() or (car.status == "pending" and car.start_date < timezone.now()):
```

Comment:
Raise a discussion here: 
Associate might create a request starting from today or next day, asking for at least 30 day entry to target account. The account admin took some time (might be two or three days) to approve it but found it expired. Have to ask the associate to update it. Back and forth.
So maybe it is tolerable to to just simply expire only the things "car.end_date < timezone.now()"?

Reply:
That's actually what I had originally, @astrozzc, but then I noticed that the AC mentioned checking the start_date too. Personally I'm ok with just checking the end_date if you and @coderbydesign are.